packages/mobile/src/screens/edit-track-screen/components/CollectibleGatedAvailability.tsx

```diff
@@ -87,9 +135,16 @@ export const CollectibleGatedAvailability = ({
     : neutral
 
   const { set: setTrackAvailabilityFields } = useSetTrackAvailabilityFields()
+  const [{ value: premiumConditions }] =
+    useField<Nullable<PremiumConditions>>('premium_conditions')
+  const nftCollection = useMemo(() => {
+    return premiumConditions?.nft_collection ?? null
+  }, [premiumConditions])
 
+  // If collectible gated was not previously selected,
+  // set as collectible gated and reset other fields.
   useEffect(() => {
-    if (selected) {
+    if (!('nft_collection' in (premiumConditions ?? {})) && selected) {
```

Comment:
we can use `nftCollection` that we just memo'd here?

Reply:
not exactly because there is temp state where we know the user wants to gate a track on an nft collection but has not yet decided which collection. in this case, we currently handle it with `{nft_collection: undefined}`, which is why we check for the key here